prot.c

```diff
@@ -1,21 +1,15 @@
-#include "hasshsgr.h"
+#include <sys/types.h>
+#include <unistd.h>
+#include <grp.h>
 #include "prot.h"
 
-/* XXX: there are more portability problems here waiting to leap out at me */
-
-int prot_gid(gid) int gid;
+int prot_gid(gid_t gid)
 {
-#ifdef HASSHORTSETGROUPS
-  short x[2];
-  x[0] = gid; x[1] = 73; /* catch errors */
-  if (setgroups(1,x) == -1) return -1;
-#else
   if (setgroups(1,&gid) == -1) return -1;
-#endif
   return setgid(gid); /* _should_ be redundant, but on some systems it isn't */
 }
 
-int prot_uid(uid) int uid;
+int prot_uid(uid_t uid)
```

Comment:
I hate to say it, but we are not there yet…

Reply:
> I hate to say it, but we are not there yet…

I had converting K&R function signatures to ANSI (C89) for the 1.08 release.  Did we put that on another release, or you're saying that we should put some 1.08 commits in before we update function signatures?

Reply:
I would at least separate the "C89 cleanup" and any functional change (and even those adding missing headers) into separate PRs, regardless if they all end up in the same release or not.